File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution{
public:
set<vector<int>> st;
void backTrack(vector<int>& nums,int i,vector<int> &temp){
    if(i>=nums.size()){
        st.insert(temp);
        return;
    }
    temp.push_back(nums[i]);
    backTrack(nums,i+1,temp);
    temp.pop_back();
    backTrack(nums,i+1,temp);

}
    vector<vector<int>> subsets(vector<int>& nums) {
           vector<int> temp;
           backTrack(nums,0,temp);
           vector<vector<int>> ans;
           for(auto v:st){
             ans.push_back(v);
           }
           return ans;
    }
    //O(N!) Not efficient , must be optimzed with dp;
};

File: Data Structures & Algorithms/baseball-game/submission-0.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        s=[]
        n=len(operations)
        for o in operations :
            if o == "C" :
                if s : s.pop()
            
            elif o == "D" :
                s.append(2*s[-1])
            
            elif o == "+" :
                if len(s)>=2:
                    a=s[-1]
                    b=s[-2]
                    s.append(a+b)
            else :
                s.append(int(o))
            
        
        ans=0
        while s :
            ans+=s.pop()
        
        return ans

        